crates/bevy_asset/src/assets.rs

```diff
@@ -549,9 +549,7 @@ impl<A: Asset> Assets<A> {
         while let Ok(drop_event) = assets.handle_provider.drop_receiver.try_recv() {
             let id = drop_event.id.typed();
 
-            assets.queued_events.push(AssetEvent::Unused { id });
-
-            let mut remove_asset = true;
+            let mut asset_unused = true;
 
             if drop_event.asset_server_managed {
                 let untyped_id = drop_event.id.untyped(TypeId::of::<A>());
```

Comment:
```suggestion
                let untyped_id = id.untyped();
```

A typed `id` is already available from line 550, so might as well use it for a cleaner API.